examples/z_pub_shm.c

```diff
@@ -65,11 +65,15 @@ int main(int argc, char **argv) {
         exit(-1);
     }
 
-    for (int idx = 0; idx < N; ++idx) {
+    for (int idx = 0; true; ++idx) {
```

Comment:
This means the loop will never exit, making it de facto a `z_pub_shm_thr` example?

Reply:
Since it prints here and there, I wouldn't call it that. The goal here was to demonstrate running the GC